.github/workflows/build-and-test.yml

```
@@ -60,11 +60,29 @@ jobs:
-      - name: Run tests with coverage
+      - name: Check if integration tests changed
+        uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            integration:
+              - 'tests/integration/**'
+              - '.github/workflows/build-and-test.yml'
+
+      - name: Run tests with coverage (including integration)
+        if: steps.filter.outputs.integration == 'true' && secrets.HEVY_API_KEY != ''
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)